test/test.c

```diff
@@ -28741,6 +28744,792 @@ void test_JetStreamSubscribePull(void)
+    testCond((s == NATS_OK) && args.msgReceived && (args.closed == false));
+    args.msgReceived = false;
+    natsMutex_Unlock(args.m);
+
+    test("Reconnect: ");
+    s = natsConnection_Flush(nc);
+    if (s == NATS_OK)
+        s = natsConnection_Reconnect(nc);
+    nats_Sleep(100);
+    testCond(s == NATS_OK);
+
+    test("Send 2 more messages: ");
+    s = js_Publish(NULL, js, "foo", "hell1", 5, NULL, &jerr);
+    if (s == NATS_OK)
+        s = js_Publish(NULL, js, "foo", "hell2", 5, NULL, &jerr);
+    testCond(s == NATS_OK);
+
+    test("Receive all expected messages: ");
+    testCond(_testBatchCompleted(&args, sub, 500, NATS_MAX_DELIVERED_MSGS, 3, false));
+
+    natsSubscription_Destroy(sub);
+    JS_TEARDOWN;
+    _destroyDefaultThreadArgs(&args);
+}
+
+void test_JetStreamSubscribePullAsync_Disconnect(void)
+{
+    natsStatus s;
+    natsOptions *opts = NULL;
+    s = natsOptions_Create(&opts);
+    if (s != NATS_OK)
+        FAIL("Unable to create options");
+    natsOptions_SetAllowReconnect(opts, false);
+    JS_SETUP_WITH_OPTS(2, 9, 2, opts);
+
+    test("Create stream: ");
+    jsErrCode jerr = 0;
+    jsStreamConfig sc;
+    jsStreamConfig_Init(&sc);
+    sc.Name = "TEST";
+    sc.Subjects = (const char *[1]){"foo"};
+    sc.SubjectsLen = 1;
+    s = js_AddStream(NULL, js, &sc, NULL, &jerr);
+    testCond((s == NATS_OK) && (jerr == 0));
+
+    test("Subscribe, expect 2 messsages, fetch size 2: ");
+    natsSubscription *sub = NULL;
+    jsOptions jsOpts;
+    struct threadArg args;
+
+    s = _createDefaultThreadArgsForCbTests(&args);
+    if (s != NATS_OK)
+        FAIL("Unable to setup test");
+
+    natsMutex_Lock(args.m);
+    args.control = 2;      // no ack
+    args.status = NATS_OK; // batch exit status will be here
+    jsOptions_Init(&jsOpts);
+    jsOpts.PullSubscribeAsync.CompleteHandler = _completePullAsync;
+    jsOpts.PullSubscribeAsync.CompleteHandlerClosure = &args;
+    jsOpts.PullSubscribeAsync.FetchSize = 2;
+    natsMutex_Unlock(args.m);
+    jsFetchRequest lifetime = {
+        .Batch = 2,
+        .Expires = NATS_SECONDS_TO_NANOS(10),
+    };
+    s = js_PullSubscribeAsync(&sub, js, "foo", "dur", _recvPullAsync, &args, &lifetime, &jsOpts, NULL, &jerr);
+    testCond(s == NATS_OK);
+
+    test("Send 1 message: ");
+    s = js_Publish(NULL, js, "foo", "hell0", 5, NULL, &jerr);
+    testCond(s == NATS_OK);
+
+    test("Receive 1 message: ");
+    natsMutex_Lock(args.m);
+    while ((s != NATS_TIMEOUT) && !args.msgReceived)
+        s = natsCondition_TimedWait(args.c, args.m, 20);
+    testCond((s == NATS_OK) && args.msgReceived && (args.closed == false));
+    args.msgReceived = false;
+    natsMutex_Unlock(args.m);
+
+    test("Disconnect: ");
+    natsConn_Lock(nc);
+    natsSock_Shutdown(nc->sockCtx.fd);
+    natsConn_Unlock(nc);
+    testCond(s == NATS_OK);
+
+    test("Check fetch completion, expect NATS_OK: ");
+    testCond(_testBatchCompleted(&args, sub, 500, NATS_OK, 1, false));
```

Comment:
@kozlovic This is rather unfortunate: the present code invokes the callback with a `NATS_OK`, rather than `NATS_CONNECTION_DISCONNECTED` status. I would appreciate a suggestion on how to best pass that context into the callback [here](https://github.com/nats-io/nats.c/pull/785/files#diff-d04d35062495ef1ffc5dedde8c86073128c37a4e0594ee1a8cd3bee7e7552729R250) and [here](https://github.com/nats-io/nats.c/pull/785/files#diff-d04d35062495ef1ffc5dedde8c86073128c37a4e0594ee1a8cd3bee7e7552729R523)

Reply:
Sorry, those links don't work for me. So not sure what you are asking really.